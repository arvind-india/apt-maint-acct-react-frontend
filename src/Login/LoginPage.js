import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormText,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions, alertActions } from '../_actions'
import { SocialLoginPage } from './SocialLoginPage'

class LoginPage extends React.Component {
  constructor(props) {
    super(props)

    // reset login status
    this.props.dispatch(userActions.logout())
    this.props.dispatch(alertActions.clear())  // clear alert messages from other pages

    this.state = {
      email: 'guest@eastgate.in',
      password: 'password123',
      submitted: false
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(e) {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }
  handleSubmit(e) {
    e.preventDefault()
    this.setState({ submitted: true })
    const { email, password } = this.state
    const { dispatch } = this.props
    if(email && password) {
      dispatch(userActions.login(email, password))
    }
  }
  render() {
    //const { email, password, submitted } = this.state
    const { alert } = this.props
    return (
      <div>
        <h2 align="center">Login</h2>
        {alert.message && <div className={`alert ${alert.type}`}>{alert.message}</div>}
        <Form onSubmit={this.handleSubmit}>
          { this.emailId() }
          { this.password() }
          { this.buttons1() }
          { this.buttons2() }
        </Form>
      </div>
    )
  }

  emailId() {
    const { email, submitted } = this.state
    return <FormGroup row>
      <Label for="email" sm={{size: 1, offset: 4}}>Email_Id</Label>
      <Col sm={3}>
        <Input
          type="email"
          name="email"
          placeholder="email id here"
          defaultValue={email}
          onChange={this.handleChange.bind(this)}
        />
        {submitted && !email && <FormText color="danger">Email-id is required</FormText>}
      </Col>
    </FormGroup>
  }

  password() {
    const { password, submitted } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>Password</Label>
      <Col sm={3}>
        <Input
          type="password"
          name="password"
          placeholder="password here"
          defaultValue={password}
          onChange={this.handleChange.bind(this)}
        />
        {submitted && !password && <FormText color="danger">Password is required</FormText>}
      </Col>
    </FormGroup>
  }

  buttons1() {
    //<Label sm={{size: 3, offset: 6}}></Label>
    return <FormGroup row>
      <Col sm={{size: 3, offset: 5}}>
        <Button type="submit" color="primary" bssize="large" title="Go to App">Login</Button>
        <Button color="link" title="Go to home">
          <Link to="/home" className="text-danger" >Cancel</Link>
        </Button>
        <Button color="link" title="Forgot Password" className="float-right">
          <Link to="/login/forgot-password" className="text-info" title="Request for resetting forgotton password">Forgot Password?</Link>
        </Button>
      </Col>
    </FormGroup>
  }

  buttons2() {
    // <Label sm={3}></Label>
    return <FormGroup row>
      <Col sm={{size: 3, offset: 5}}>
        <Button color="link" title="Go to Registration" className="float-left">
          <Link to="/register" className="text-success">New User?</Link>
        </Button>
        <SocialLoginPage />
      </Col>
    </FormGroup>
  }

}

function mapStateToProps(state) {
  const { alert } = state
  const { loggingIn } = state.authentication
  return {
    alert,
    loggingIn
  }
}

const connectedLoginPage = connect(mapStateToProps)(LoginPage)
export { connectedLoginPage as LoginPage }
