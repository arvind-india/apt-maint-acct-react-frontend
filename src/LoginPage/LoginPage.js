import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormText,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions } from '../_actions'

class LoginPage extends React.Component {
  constructor(props) {
    super(props)

    // reset login status
    this.props.dispatch(userActions.logout())

    this.state = {
      email: 'guest@eastgate.in',
      password: 'password123',
      submitted: false
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(e) {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }
  handleSubmit(e) {
    e.preventDefault()

    this.setState({ submitted: true })
    const { email, password } = this.state
    const { dispatch } = this.props
    if(email && password) {
      dispatch(userActions.login(email, password))
    }
  }
  render() {

    const { email, password, submitted } = this.state
    const { alert } = this.props

    return (
      <div>
        <h2 align="center">Login</h2>
        {alert.message && <div className={`alert ${alert.type}`}>{alert.message}</div>}

        <Form onSubmit={this.handleSubmit}>

          <FormGroup row>
            <Label for="email" sm={3}>Email_Id</Label>
            <Col sm={9}>
              <Input
                type="email"
                name="email"
                placeholder="email id here"
                defaultValue={email}
                onChange={this.handleChange.bind(this)}
              />
              {submitted && !email && <FormText color="danger">Email-id is required</FormText>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>Password</Label>
            <Col sm={9}>
              <Input
                type="password"
                name="password"
                placeholder="password here"
                defaultValue={password}
                onChange={this.handleChange.bind(this)}
              />
              {submitted && !password && <FormText color="danger">Password is required</FormText>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}></Label>
            <Col smoffset={3} sm={9}>
              <Button type="submit" color="primary" bssize="large">Login</Button>
              <Button color="link">
                <Link to="/register">Register</Link>
              </Button>
            </Col>
          </FormGroup>

        </Form>
      </div>
    )
  }
}

function mapStateToProps(state) {
  const { alert } = state
  const { loggingIn } = state.authentication
  return {
    alert,
    loggingIn
  }
}

const connectedLoginPage = connect(mapStateToProps)(LoginPage)
export { connectedLoginPage as LoginPage }
