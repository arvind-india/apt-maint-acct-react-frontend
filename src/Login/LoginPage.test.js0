// LoginPage.test.js

import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'

import { Login } from './LoginPage'


// pass in middleware if needed into configureStore
const mockStore = configureMockStore()

describe('Login Component', () => {
  let wrapper
  // let store
  const mockLoginFn = jest.fn()
  const auth = {loggingIn: true}
  const state = {
    alert: "",
    authentication: ""
  }

  beforeEach(() => {
    const initialState = {
      alert: {message: "testingAlert"},
      authentication: {loggingIn: true, user: {}},
      login: mockLoginFn,
      logout: jest.fn(), // mock function
      clearAlert: jest.fn()
    }
    let store = mockStore(initialState)
    // creates the store with any initial state or middleware needed
     wrapper = shallow(
      <Login store={store} />
     )
    //  wrapper = shallow(<Login {...initialState} />)
      console.log('Wrapper: ', wrapper.props())
  })

  it('should show alert message', () => {
    expect(wrapper.props().alert.message).toBe('testingAlert')
  })

  describe('When the form is submitted', () => {
    it('should call the mock login function', () => {
      wrapper.find('#loginForm').simulate(
        'submit',
        {preventDefault() {}}
      )
      expect(mockLoginFn.mock.calls.length).toBe(1)
    })

    it('should be called with the email and password in the state as arguments', () => {
      wrapper.find('#email').simulate(
        'change',
        {target: {name: 'email', value: "dummy@email.id"}}
      )
      wrapper.find('#password').simulate(
        'change',
        {target: {name: 'password', value: "monkeys"}}
      )
      wrapper.find('#loginForm').simulate(
        'submit',
        {preventDefault() {}}
      )
      // test to see arguments used after its been submitted
      expect(mockLoginFn.mock.calls[1][0]).toEqual(
        {email: 'dummy@email.id', password: 'monkeys'}
      )
    })
  })

})
