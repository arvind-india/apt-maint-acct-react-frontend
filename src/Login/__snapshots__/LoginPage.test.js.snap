// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage params test should render initial layout 1`] = `
Array [
  <div>
    <h2
        align="center"
    >
        Login
    </h2>
    <FlashMessage
        color={undefined}
        delay={4000}
        text="testingAlert"
    />
    <Form
        id="loginForm"
        onSubmit={[Function]}
        tag="form"
    >
        <FormGroup
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="email"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Email id
                </Label>
                <Input
                    defaultValue="guest@eastgate.in"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    tag="p"
                    title="email id here"
                    type="email"
                />
            </Col>
        </FormGroup>
        <FormGroup
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="password"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Password
                </Label>
                <Input
                    defaultValue="password123"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    tag="p"
                    title="password here"
                    type="password"
                />
            </Col>
        </FormGroup>
        <FormGroup
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Button
                    bssize="large"
                    color="primary"
                    disabled={false}
                    id="submitButton"
                    tag="button"
                    title="Save Changes"
                    type="submit"
                >
                    Login
                </Button>
                <Button
                    color="link"
                    tag="button"
                    title="Go to home"
                >
                    <Link
                        className="text-danger"
                        replace={false}
                        to="/home"
                    >
                        Cancel
                    </Link>
                </Button>
                <Connect(SocialLogin) />
            </Col>
        </FormGroup>
        <FormGroup
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Button
                    className="float-right"
                    color="link"
                    id="newUser"
                    tag="button"
                    title="Go to Registration"
                >
                    <Link
                        className="text-success"
                        replace={false}
                        to="/register"
                    >
                        New User?
                    </Link>
                </Button>
                <Button
                    className="float-left"
                    color="link"
                    id="forgotPassword"
                    tag="button"
                    title="Forgot Password"
                >
                    <Link
                        className="text-info"
                        replace={false}
                        title="Request for resetting forgotton password"
                        to="/login/forgot-password"
                    >
                        Forgot Password?
                    </Link>
                </Button>
            </Col>
        </FormGroup>
    </Form>
</div>,
]
`;
